pkg/errors/error_codes.go

```diff
@@ -45,8 +45,12 @@ const (
 )
 
 func contextToString(ctx context.Context) string {
-	contextType := reflect.TypeOf(ctx).Elem()
-	if contextType.Kind() != reflect.Struct {
+	typ := reflect.TypeOf(ctx)
+	if fmt.Sprintf("XXX%s", typ) == "XXXcontext.backgroundCtx" {
```

Comment:
just wondering, what's the point of prepending "XXX" here? Why not do `if fmt.Sprintf("%s", typ) == "context.backgroundCtx" {}`?

Reply:
makes it easier when logging if the string is empty.